src/main/java/com/faforever/server/integration/legacy/LegacyServicesActivators.java

```diff
@@ -51,6 +54,10 @@ public SessionResponse askSession() {
   @ServiceActivator(inputChannel = ChannelNames.LEGACY_LOGIN_REQUEST)
   @Transactional(readOnly = true)
   public void loginRequest(LoginMessage loginRequest, @Header(CLIENT_CONNECTION) ClientConnection clientConnection) {
+    if (playerService.isPlayerOnline(loginRequest.getLogin())) {
+      throw new RequestException(ErrorCode.USER_ALREADY_CONNECTED);
+    }
```

Comment:
This should be unit tested. Also, you can use `Requests.verify(!playerService.isPlayerOnline(loginRequest.getLogin(), ErrorCode.USER_ALREADY_CONNECTED);`